Source/Diagnostics/BTDiagnostics.cpp

```diff
@@ -662,6 +662,12 @@ void BTDiagnostics::TMP_ClearSpeciesDataForBTD ()
 
 void BTDiagnostics::MergeBuffersForPlotfile (int i_snapshot)
 {
+    // Make sure all MPI ranks wrote their files and closed it
+    // Note: additionally, since a Barrier does not guarantee a FS sync
+    //       on a parallel FS, we might need to add timeouts and re-tries
+    //       to the open calls below when running at scale.
+    amrex::ParallelDescriptor::Barrier();
```

Comment:
@RevathiJambunathan are we guaranteeing that all MPI ranks reach this point? :)
Otherwise we could call a barrier on a sub-communicator, if needed.

Reply:
All the I/O for BTD is done within `amrex:ParallelContext::IOProcessorSub()` for the cases when we have only some processors dedicated to I/O

Reply:
There is also `amrex::ParallelContext::BarrierSub()` for that.

Reply:
Ok, cool!

So that means the current PR proposal should work (sync all that potentially wrote a plotfile).

Btw, is there a specific reason for the barrier at the end of this function?